message_definitions/v1.0/ras_a.xml

```diff
@@ -16,6 +16,67 @@
         <param index="2" label="X Position in the camera frame" minValue="0" increment="1">Position given by the number of pixels in X direction of the camera frame. Origin on the bottom left corner of the camera frame.</param>
         <param index="3" label="Y Position in the camera frame" minValue="0" increment="1">Position given by the number of pixels in Y direction of the camera frame. Origin on the bottom left corner of the camera frame.</param>
       </entry>
+      <entry value="36" name="MAV_CMD_DO_FOLLOW_SYS_ID" hasLocation="false" isDestination="false">
+        <description>Begin following a target</description>
+        <param index="1" label="System ID" minValue="1" maxValue="255" increment="1">System ID (of the FOLLOW_TARGET beacon).</param>
+        <param index="2">Reserved</param>
+        <param index="3">Reserved</param>
+        <param index="4" label="Altitude Mode" minValue="0" maxValue="2" increment="1">Altitude mode: 0: Keep current altitude, 1: keep altitude difference to target, 2: go to a fixed altitude above home.</param>
+        <param index="5" label="Altitude" units="m">Altitude above home. (used if mode=2)</param>
+        <param index="6">Reserved</param>
+        <param index="7" label="Time to Land" units="s" minValue="0">Time to land in which the MAV should go to the default position hold mode after a message RX timeout.</param>
+      </entry>
+      <entry value="37" name="MAV_CMD_DO_FOLLOW_POI" hasLocation="false" isDestination="false">
+        <description>Begin following a target specified by POI ID.  A POI Unique ID (UID) is a 64-bit field, and so to fully represent that UID, the big endian 64-bit POI UID is split into 2 words; param1: upper 32-bit word, param2: lower 32-bit word.</description>
+        <param index="1" label="POI UID Upper 32 Bits">The upper 32 bits of the UID from a POI_REPORT.</param>
+        <param index="2" label="POI UID Lower 32 Bits">The lower 32 bits of the UID from a POI_REPORT</param>
+        <param index="3" label="Mode Flags" enum="FOLLOW_POI_MODE_FLAGS">Bitmask of option flags.</param>
+        <param index="4" label="Altitude" units="mm or degE7">Altitude field, behavior to be interpreted according to Mode Flags.</param>
```

Comment:
FYI only: it is an anti-pattern (not recommended if possible) to have one field with two possible meetings. Yes I know you see heaps of that in the old messages, but we try and avoid it on new messages where possible.

Reply:
To put it another way, do you really _need_ to specify the altitude, range, etc in multiple ways? (what is the benefit)?

Reply:
indeed I did take inspiration probably from older messages. but also working in the constraints of a MAV_CMD, it was the only option short of creating ridiculous permutations to represent different fields. Any mix of the mode flags provided is reasonable. assuming the AP supports them, reduces required bandwidth and complexity on the client side. some of them map to similar MAV_FRAME(s), but we couldn't just assume that 3 fields should use same MAV_FRAME.